c7n/resources/rds.py

```diff
@@ -1568,3 +1568,105 @@ def process(self, resources, event=None):
+                  - type: modify-db
+                    update:
+                        - keyword: 'DeletionProtection'
+                            value: false
+                        - keyword: 'PubliclyAccessible'
+                            value: false
+                    immediate: true
+    """
+
+    schema = type_schema(
+        'modify-db',
+        immediate={"type": 'boolean'},
+        update={
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'properties': {
+                    'keyword': {'type': 'string', 'enum': [
+                        'AllocatedStorage',
+                        'DBInstanceClass',
+                        'DBSubnetGroupName',
+                        'DBSecurityGroups',
+                        'VpcSecurityGroupIds',
+                        'MasterUserPassword',
+                        'DBParameterGroupName',
+                        'BackupRetentionPeriod',
+                        'PreferredBackupWindow',
+                        'PreferredMaintenanceWindow',
+                        'MultiAZ',
+                        'EngineVersion',
+                        'AllowMajorVersionUpgrade',
+                        'AutoMinorVersionUpgrade',
+                        'LicenseModel',
+                        'Iops',
+                        'OptionGroupName',
+                        'NewDBInstanceIdentifier',
+                        'StorageType',
+                        'TdeCredentialArn',
+                        'TdeCredentialPassword',
+                        'CACertificateIdentifier',
+                        'Domain',
+                        'CopyTagsToSnapshot',
+                        'MonitoringInterval',
+                        'DBPortNumber',
+                        'PubliclyAccessible',
+                        'DomainIAMRoleName',
+                        'PromotionTier',
+                        'EnableIAMDatabaseAuthentication',
+                        'EnablePerformanceInsights',
+                        'PerformanceInsightsKMSKeyId',
+                        'PerformanceInsightsRetentionPeriod',
+                        'CloudwatchLogsExportConfiguration',
+                        'UseDefaultProcessorFeatures',
+                        'DeletionProtection']},
+                    'value': {}
+                },
+            },
+        },
+        required=('update',))
+
+    permissions = ('rds:ModifyDBInstance',)
+
+    def process(self, resources):
+        c = local_session(self.manager.session_factory).client('rds')
+
+        param = {}
+        param['ApplyImmediately'] = self.data.get('immediate', False)
+
+        for x in self.data.get('update'):
+            param[x['keyword']] = x['value']
+
+        for r in resources:
+            param['DBInstanceIdentifier'] = r['DBInstanceIdentifier']
+            try:
+                c.modify_db_instance(**param)
+            except ClientError as e:
```

Comment:
catch not found, else this should be raising exceptions. exceptions shouldn't be silently passing unless explicitly silenced and log is equivalent to silent for unattended operations.

Reply:
Switched raising an exception on e.response['Error']['Code'] != 'DBInstanceNotFound'. 

https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ModifyDBInstance.html#API_ModifyDBInstance_Errors

Reply:
rather than catch on ClientError try to use the client exceptions as it results in less error handling code or string  comparison, ie.

`except client.exceptions.DBInstanceNotFoundFault:`

Reply:
Made the change, I'm curious how you know which exceptions boto3 has for each resource. I can see the AWS language non-specific documentation for it on the link I posted above. For Python specific errors it links to the boto3 documentation which doesn't really give a list of possible exceptions. I did some diving and found this 

https://github.com/boto/boto3/issues/1262

I was wondering if printing the keys in the dict is the best possible way to view the possible exceptions that boto3 has, 

rds.exceptions.__dict__['_code_to_exception'].keys()

Reply:
I typically do this via console.. they are generated so its not per se easy to discover via docs, and its also not super consistent imo across services over which are reflected back.

```
$ python
>>> client = boto3.client('service')
>>>for i in sorted(dir(client)): print i
```